frontend/packages/core/src/AppProvider/workflow-hydrator.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import type { clutch as IClutch } from "@clutch-sh/api";
+import { Grid } from "@material-ui/core";
+
+import { WorkflowStorageContext } from "../Contexts/workflow-storage-context";
+import { retrieveData } from "../Contexts/workflow-storage-context/helpers";
+import workflowStorageContextReducer from "../Contexts/workflow-storage-context/reducer";
+import type { WorkflowStorageContextProps } from "../Contexts/workflow-storage-context/types";
+import { defaultWorkflowStorageState } from "../Contexts/workflow-storage-context/types";
+import { Alert } from "../Feedback";
+import styled from "../styled";
+
+interface WorkflowHydratorProps {
+  children: React.ReactElement;
+  hydrateData: IClutch.shortlink.v1.IShareableState[];
+  onClear: () => void;
+}
+
+const StyledAlert = styled(Alert)({
+  zIndex: 1200,
+  position: "absolute",
+  padding: "6px 8px",
+});
+
+/**
+ * Hydrator which is a wrapper for workflows
+ * Will check on load if there exists any hydrated data for the current workflow
+ * If there is it will populate the state and provide an alert above the workflow
+ */
+const WorkflowHydrator = ({
+  hydrateData,
+  onClear,
+  children,
+}: WorkflowHydratorProps): React.ReactElement => {
+  const [state, dispatch] = React.useReducer(
+    workflowStorageContextReducer,
+    defaultWorkflowStorageState
+  );
+
+  React.useEffect(() => {
+    if (hydrateData && hydrateData.length) {
+      dispatch({ type: "HYDRATE", payload: { data: hydrateData } });
+      onClear();
+    }
+  }, [hydrateData]);
+
+  const storageProviderProps: WorkflowStorageContextProps = {
+    shortLinked: state.shortLinked,
```

Comment:
nit: I think this is to denote the entry was from a short link, if so how about
```suggestion
    fromShortLink: state.fromShortLink,
```

Reply:
Yeah its meant to denote that the data has been loaded via a shortlink, in case anything needs to be modified for a component directly or any different actions need to be taken (ie showing the alert banner). I can rename it though